src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs

```diff
@@ -75,6 +75,7 @@ internal enum PreconditionState
             if (preconditionState == PreconditionState.NotModified)
             {
                 response.StatusCode = StatusCodes.Status304NotModified;
+                response.Headers.Remove("Content-Type");
```

Comment:
@Tratcher does a 412 require the `Content-Type` header to be sent? If not, we could move these two lines https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs#L56-L57 to before here - https://github.com/aspnet/AspNetCore/blob/master/src/Mvc/Mvc.Core/src/Infrastructure/FileResultExecutorBase.cs#L86.

Reply:
No, 412 doesn't need it either. Moving SetContentType and SetContentDispositionHeader would be better than removing the value here.

Reply:
Done